datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -478,6 +489,10 @@ protected void trackEventsProcessedProgress(int recordCount) {
       }
       _consumerMetrics.updateNumPolls(1);
       _consumerMetrics.updateEventCountsPerPoll(records.count());
+      if (_enablePollDurationMsMetric) {
+        _dynamicMetricsManager.createOrUpdateHistogram(_metricsPrefix, _datastreamName, POLL_DURATION_MS,
```

Comment:
Do you think refactoring `createKafkaBasedConnectorTaskMetrics` with additional boolean which decides the creation of histogram for poll duration will make more sense? Then in the future, if we decide to enable it for other tasks, we can just set the boolean to true?

Reply:
So the problem is that even if I use the boolean in `createKafkaBasedConnectorTaskMetrics` to create the metric, I'll have to update the metric here, and update it under the same boolean. So I thought why not just create the metric lazily. Plus, this metric lives in `AbstractKafkaBasedConnectorTask`, which means that any connectors that extend this class will get this metric already. Each connector can also decide they don't want this metric, and disable it via config.

Do let me know if I misunderstood your suggestion, and if you're okay with me not adding code to register this metric in `createKafkaBasedConnectorTaskMetrics`.

Reply:
Resolving after offline discussion.